newrelic/structures_newrelic_alert_muting_rule.go

```diff
@@ -16,10 +20,170 @@ func expandMutingRuleCreateInput(d *schema.ResourceData) alerts.MutingRuleCreate
 		createInput.Condition = expandMutingRuleConditionGroup(e.([]interface{})[0].(map[string]interface{}))
 	}
 
-	return createInput
+	if e, ok := d.GetOk("schedule"); ok {
+		schedule, err := expandMutingRuleCreateSchedule(e.([]interface{})[0].(map[string]interface{}))
+		if err != nil {
+			return alerts.MutingRuleCreateInput{}, err
+		}
+		createInput.Schedule = &schedule
+	}
+
+	return createInput, nil
+}
+
+func expandMutingRuleCreateSchedule(cfg map[string]interface{}) (alerts.MutingRuleScheduleCreateInput, error) {
+	schedule := alerts.MutingRuleScheduleCreateInput{}
+
+	if startTime, ok := cfg["start_time"]; ok {
+		rawStartTime := startTime.(string)
+		if rawStartTime != "" {
+			formattedStartTime, err := time.Parse("2006-01-02T15:04:05", rawStartTime)
```

Comment:
I'm a little confused why we need `time.Parse("2006-01-02T15:04:05", rawStartTime)`.  Haven't we already validated this is in a naive date time format using the `validateNaiveDateTime` function in `resource_newrelic_alert_muting_rule.go`?

Reply:
The validator will catch incorrectly formatted date strings before attempting to prepare a payload.. the validateNaiveDateTime function takes a time.time, converts it and then returns the errors while throwing away the result.
In this case, we do the conversion for real, going from a go time.time to a string.

Reply:
👍 I keep forgetting we provide a `time.time` type that needs to be converted to a `string`.  For some reason when I wrote this question, I was thinking we provide a `string` and `time.Parse` just reformats that `string`.  My bad.  Thanks!